legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -160,10 +143,17 @@ public Iterable<? extends Processor<?>> getExtraProcessors()
 
                     // executables
                     metamodel._executables(dataSpace.executables != null ? ListIterate.collect(dataSpace.executables, executable ->
-                            new Root_meta_pure_metamodel_dataSpace_DataSpaceExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpaceExecutable"))
-                                    ._title(executable.title)
-                                    ._description(executable.description)
-                                    ._executable(context.pureModel.getPackageableElement(executable.executable.path, executable.executable.sourceInformation))) : null);
+                            executable instanceof DataSpacePackageableElementExecutable ?
```

Comment:
for `we should also add a check that the executable's title is unique.`, 
To be backward compatible, I just did the check for template queries since we don't check duplicated titles in the original flow.

Reply:
can you query metadata using classifier apis to check if there are many datapsaces with more than one executables. dont anticipate many of them using executables and even more using the same titles. 
If its only a handful im okay with breaking. I would guess there are none with multi titles that are the same. 
I would prefer to add this compiler check for all.

Reply:
I checked the metadata, all current titles are unique. 
Moved the logic to check all titles